ipatests/test_integration/test_ipahealthcheck.py

```diff
@@ -904,3 +910,101 @@ def test_ipahealthcheck_trust_agent_member_check(self):
+        )
+
+        install_server_external_ca_step2(
+            cls.master, ipa_ca_fname, root_ca_fname
+        )
+        tasks.kinit_admin(cls.master)
+
+    def test_ipahealthcheck_crlmanagercheck(self):
+        """
+        Test for IPACRLManagerCheck
+        """
+        returncode, data = run_healthcheck(
+            self.master, "ipahealthcheck.ipa.roles", "IPACRLManagerCheck"
+        )
+        assert returncode == 0
+        for check in data:
+            assert check["result"] == "SUCCESS"
+            assert check["kw"]["key"] == "crl_manager"
+            assert check["kw"]["crlgen_enabled"] is True
+
+    def test_ipahealthcheck_certmongerca(self):
+        """
+        Test for IPACertmongerCA
+        """
+        returncode, data = run_healthcheck(
+            self.master, "ipahealthcheck.ipa.certs", "IPACertmongerCA",
+        )
+        assert returncode == 0
+        for check in data:
+            assert check["result"] == "SUCCESS"
+            assert (
+                check["kw"]["key"] == "IPA"
+                or check["kw"]["key"] == "dogtag-ipa-ca-renew-agent"
+                or check["kw"]["key"] == "dogtag-ipa-ca-renew-agent-reuse"
+            )
+
+    def test_ipahealthcheck_ipaopensslchainvalidation(self):
+        """
+        Test for IPAOpenSSLChainValidation
+        """
+        returncode, data = run_healthcheck(
+            self.master,
+            "ipahealthcheck.ipa.certs",
+            "IPAOpenSSLChainValidation",
+        )
+        assert returncode == 0
+        for check in data:
+            assert check["result"] == "SUCCESS"
+            assert (
+                check["kw"]["key"] == paths.HTTPD_CERT_FILE
+                or check["kw"]["key"] == paths.RA_AGENT_PEM
+            )
+
+    def test_ipahealthcheck_ipansschainvalidation(self):
+        """
+        Test for IPANSSChainValidation
+        """
+        instance = realm_to_serverid(self.master.domain.realm)
+        returncode, data = run_healthcheck(
+            self.master, "ipahealthcheck.ipa.certs", "IPANSSChainValidation",
+        )
+        assert returncode == 0
+        for check in data:
+            if check["kw"]["dbdir"] == (paths.PKI_TOMCAT_ALIAS_DIR):
+                assert check["result"] == "SUCCESS"
+                assert check["kw"]["nickname"] == "Server-Cert cert-pki-ca"
+            elif check["kw"]["dbdir"] == (
+                paths.ETC_DIRSRV_SLAPD_INSTANCE_TEMPLATE % instance
+            ):
+                assert check["result"] == "SUCCESS"
+                assert check["kw"]["nickname"] == "Server-Cert"
+
+    def test_ipahealthcheck_iparaagent(self):
+        """
+        Test for IPARAAgent
+        """
+        returncode, data = run_healthcheck(
+            self.master, "ipahealthcheck.ipa.certs", "IPARAAgent"
+        )
+        assert returncode == 0
+        for check in data:
+            assert check["result"] == "SUCCESS"
+            assert check["kw"] == {}
```

Comment:
For completion sake I guess this is ok but this is enforced by the baseline test as well so if that fails this will fail too.

This test has a lot of room to test things that can go wrong:

1. paths.RA_AGENT_PEM is missing or mal-formatted
2. The entry doesn't exist in LDAP at all
3. The description doesn't match expectations (like differing serial #)
4. The userCertificate value is different or missing

Reply:
> For completion sake I guess this is ok but this is enforced by the baseline test as well so if that fails this will fail too.
> 
> This test has a lot of room to test things that can go wrong:
> 
>     1. paths.RA_AGENT_PEM is missing or mal-formatted
> 
>     2. The entry doesn't exist in LDAP at all
> 
>     3. The description doesn't match expectations (like differing serial #)
> 
>     4. The userCertificate value is different or missing
have replaced the existing code and  implemented one of the scenario to modify the description of the RA cert in ldap and the tool reports the status as "RA agent not found in LDAP."